### ISSUE TITLE
Qt5/Windows 11? Large (plugin) dialogs disappear unrecovarable (if modal blocking Windows)

### ISSUE THREAD

**Issue by rduivenvoorde** [qgis/QGIS#47165]
### What is the bug or the crash?

See also: https://lists.osgeo.org/pipermail/qgis-developer/2022-January/064471.html

I have a plugin which has a non-resizable (potentially giant) Dialog popping up on QGIS. All working fine on large screens and all Operating Systems.

But when a friend, running Windows 11 on a laptop with a 1920x1080 screen opened the dialog (which is probably higher then 1080px!) all seems fine UNTILL he tried to reach the bottom of the dialog. He tried to do that by 'resizing' the dialog via dragging the top of the dialog down: 
- The Dialog disappears without a trace ... not in 'alt-tab', not in Taskbar...
- If I created the dialog modal, it was even blocking QGIS making it un-usable

As a work around I used a scrollable area for a group, and as soon as the Dialog is resizable it seems not an issue anymore...

### Steps to reproduce the issue

I created a (non resizable) giant dialog with a lot widgets here: https://github.com/rduivenvoorde/win11-plugin

You can download the plugin as zip https://github.com/rduivenvoorde/win11-plugin/archive/refs/heads/main.zip and install that plugin via 'Install via ZIP' in QGIS

Here in Windows 10 (in a VM on my laptop), I can resize and move the dialog without issues.

![Screenshot-20220203094755-2009x1284](https://user-images.githubusercontent.com/731673/152309937-22fbca36-fa66-40d9-96f5-ed287d89d74f.png)

But doing the same on Windows 11: the dialog moves into the bottom and is never seen again.

I'm planning to create a VM of Win11 too (https://www.tecklyfe.com/how-to-create-a-windows-11-virtual-machine-in-qemu/), but.... need time and space...

As I cannot do/test it myself here, it would be nice if a Windows 11 user could try to reproduce this...

### Versions

Currently only in Windows 11

It IS possible that it has something todo with the Windows-behaviour?

It could also be a potential Qt5/Win11 issue?

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by Houska1** [comment]
Confirmed on (native) Windows 11, with OSGeo QGIS 3.23 master, Qt 5.15.2.

**Comment by rduivenvoorde** [comment]
Yeah, that is something I'm afraid of... and as Qt does not support Qt5 anymore (?) I hope that this does not bite too many people... As said, the work-around is to make it possible to resize the window by using a scrolled area or so...

**Comment by nyalldawson** [comment]
@m-kuhn @3nids here's a good example of what we were discussing in the python working group (relating to the motivation to get working qt6 builds of qgis 3.x)

**Comment by m-kuhn** [comment]
Is this confirmed as fixed with qt6?

**Comment by nyalldawson** [comment]
@m-kuhn no, not confirmed

**Comment by AlisterH** [comment]
You might be able to get rid of the modal blocking window by pressing the Escape key (as per #46781; I wonder if these issues could be related)
